### PR TITLE
Use `keepdims` and maintain backward compatible for `keep_dims`

### PR DESCRIPTION
**Pull request by yongtang**
This fix tries to address the issue raised in #6815 where
both `keepdims` and `keep_dims` were used with inconsistency.

This fix changes related api to `keepdims` while at the same
time maintain backward compatible for `keep_dims` so that
use will not be impacted.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow/python/ops/linalg_ops.py [hunk 1]
```diff
@@ -422,7 +422,7 @@ def svd(tensor, full_matrices=False, compute_uv=True, name=None):
 
 
 # pylint: disable=redefined-builtin
-def norm(tensor, ord='euclidean', axis=None, keep_dims=False, name=None):
+def norm(tensor, ord='euclidean', axis=None, keepdims=None, name=None, keep_dims=None):
```
Comment (martinwicke) [inline tensorflow/python/ops/linalg_ops.py]: please use @deprecated_args

**Comment by yongtang** [comment]
Thanks @martinwicke for the review. The PR has been updated with `@deprecated_args` used.

**Comment by drpngx** [comment]
Any reason to choose `keepdims` instead of `keep_dims`? The latter seems like the better choice to me.

**Comment by yongtang** [comment]
@drpngx Not knowing enough history about the code base though I would assume the reason of having `keepdims` instead of `keep_dims` is to align with `numpy`? e.g., 

https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.amin.html#numpy.amin

At the moment `keras` backend use `keepdims` while the rest of tensorflow mostly use `keep_dims`.

**Comment by drpngx** [comment]
Thank you @yongtang , that makes sense, `keepdims` seems good then.

**Comment by drpngx** [comment]
So, sorry to be dense, but does that mean that current code will issue a warning when `tf.reduce_sum(.., keep_dims=True)` will issue a warning but and behave like `tf.reduce_sum(..., keepdims=False)` (the default)?

Don't we want to keep **both** and override `keep_dims` with `keepdims` if both are specified?

**Comment by yongtang** [comment]
Thanks @drpngx. The current behavior of this PR is to show a warning in doc string (from the annotation `@deprecated_args`), and accept either `keep_dims` or `keepdims` in the call.

Both `keep_dims` or `keepdims` should work.

However, in case both `keep_dims` and `keepdims` are specified in the same call, then an error will be thrown.

**Comment by drpngx** [comment]
Oh, just saw the code. Sounds good.

**Comment by drpngx** [comment]
Jenkins, test this please.

**Comment by yongtang** [comment]
The PR has been updated with 
```
    bazel-bin/tensorflow/tools/api/tests/api_compatibility_test \
              --update_goldens True
```
so that `//tensorflow/tools/api/tests:api_compatibility_test` could pass.

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by yongtang** [comment]
The `Linux CPU Tests Makefile` test might be unrelated I think?

**Comment by drpngx** [comment]
this one seems like a download error

Jenkins, test this please.

**Review by martinwicke — APPROVED** [review]

**Comment by neilpanchal** [comment]
Violates PEP guidelines, https://github.com/tensorflow/tensorflow/issues/6815#issuecomment-355110926

I vote for keeping `keep_dims`. There needs to be a strong compelling reason to change. Or support both and get rid of the annoying warnings.

**Pull request merged**